src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -378,24 +378,11 @@ private static void GetWindowSize(out int width, out int height)
 
         public static void SetWindowSize(int width, int height)
         {
-           lock (Console.Out)
-           {
-               Interop.Sys.WinSize winsize = default;
-               winsize.Row = (ushort)height;
-               winsize.Col = (ushort)width;
-               if (Interop.Sys.SetWindowSize(in winsize) == 0)
-               {
-                   s_windowWidth = winsize.Col;
-                   s_windowHeight = winsize.Row;
-               }
-               else
-               {
-                   Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
-                   throw errorInfo.Error == Interop.Error.ENOTSUP ?
-                       new PlatformNotSupportedException() :
-                       Interop.GetIOException(errorInfo);
-               }
-           }
+            // note: We can't implement SetWindowSize using TIOCSWINSZ.
+            // TIOCSWINSZ is meant to inform the kernel of the terminal size.
+            // The window that shows the terminal doesn't change to match that size.
+
+            throw new PlatformNotSupportedException();
```

Comment:
What behavior would it have before? Just nop'ing? If so, feels problematic to start throwing. Or was it already throwing?

Reply:
This throws, as was the case before .NET 8.

I think the rationale for throwing is to avoid that a user assumes that the window size changed.

Reply:
At a minimum we'll need to call it out as a breaking change.